### PR TITLE
More error handling for aria2 cmd

### PR DESCRIPTION
**Pull request by jetownfeve21**
- error handling for aria2 commands and curl
- add get aria2 launch command function

### CODE REVIEW

**Comment by kelson42** [comment]
@jetownfeve21 This is always printing things, not only in case of error like requested.

**Comment by jetownfeve21** [comment]
@kelson42 To be sure to really understand what is going on, add debug throughout the aria2 startup process seems to be more efficient for me. Maybe in a log file, it's better

**Comment by kelson42** [comment]
@jetownfeve21 If something goes wrong with aria2c cmd, it should go through one error reporting. At this stage you print the error. If everything goes right, nobody is interested to get this command line,

File: src/aria2.cpp [hunk 1]
```diff
@@ -80,16 +80,27 @@ Aria2::Aria2():
...
 
   int watchdog = 50;
   while(--watchdog) {
     sleep(10);
+    errbuf[0] = 0;
     auto res = curl_easy_perform(mp_curl);
     if (res == CURLE_OK) {
       break;
+    } else if (watchdog == 1) {
+        std::cerr << "try " << 50 - watchdog <<" curl_easy_perform() failed." << std::endl;
```
Comment (kelson42) [inline src/aria2.cpp]: Take care to the spaces

Comment (kelson42) [reply src/aria2.cpp]: Use `watchdog` instead of hardcoded `50`.

File: src/downloader.cpp [hunk 2]
```diff
@@ -127,17 +127,24 @@ void Download::cancelDownload()
...
-  for (auto gid : mp_aria->tellActive()) {
-    m_knownDownloads[gid] = std::unique_ptr<Download>(new Download(mp_aria, gid));
-    m_knownDownloads[gid]->updateStatus();
+  try {
+    for (auto gid : mp_aria->tellActive()) {
+      m_knownDownloads[gid] = std::unique_ptr<Download>(new Download(mp_aria, gid));
+      m_knownDownloads[gid]->updateStatus();
+    }
+  } catch (std::exception& e) {
+    std::cout << "aria2 tellActive failed : " << e.what();
```
Comment (kelson42) [inline src/downloader.cpp]: `stderr`

File: src/downloader.cpp [hunk 3]
```diff
@@ -127,17 +127,24 @@ void Download::cancelDownload()
...
-  for (auto gid : mp_aria->tellWaiting()) {
-    m_knownDownloads[gid] = std::unique_ptr<Download>(new Download(mp_aria, gid));
-    m_knownDownloads[gid]->updateStatus();
+  try {
+    for (auto gid : mp_aria->tellWaiting()) {
+      m_knownDownloads[gid] = std::unique_ptr<Download>(new Download(mp_aria, gid));
+      m_knownDownloads[gid]->updateStatus();
+    }
+  } catch (std::exception& e) {
+    std::cout << "aria2 tellWaiting failed : " << e.what();
```
Comment (kelson42) [inline src/downloader.cpp]: `stderr`

**Comment by kelson42** [comment]
@jetownfeve21 I'm not sure you are over with this PR, but could you please rebase and ask for review when/if you are?

**Pull request merged**